### PR TITLE
Drexel und Weiss: Add smartmeterconsumer interface

### PR DESCRIPTION
**Pull request by Boernsman**
nymea-plugins-modbus pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

### CODE REVIEW

File: drexelundweiss/integrationplugindrexelundweiss.cpp [hunk 1]
```diff
@@ -315,275 +315,317 @@ void IntegrationPluginDrexelUndWeiss::updateStates(Thing *thing)
...
+                    case ModbusRegisterX2::StoerungWertNichtZulaessig:
+                    if (value != 0)
                     thing->setStateValue(x2wpErrorStateTypeId, "Value not allowed");
-                break;
+                    break;
 
-            case ModbusRegisterX2::StoerungZuluftventilator:
-                if (value != 0)
+                    case ModbusRegisterX2::StoerungZuluftventilator:
+                    if (value != 0)
```
Comment (mzanetti) [inline drexelundweiss/integrationplugindrexelundweiss.cpp:550]: Indendation on those doesn't look right...

**Review by mzanetti — APPROVED** [review]
Nice, thanks!

Indentation looks wrong on the long swtich/case statement, perhaps you want to clean that up. Otherwise looks good.

Just FYI: There's a modbus tool available by now, where you would define the registers (and blocks) in a json and it would generate a class that takes care of all the reading and polling. No need to change this now, just as a hint if you'd touc some of the modbus plugins again.

File: drexelundweiss/integrationplugindrexelundweiss.cpp (hunk 1 above)
Comment (Boernsman) [reply drexelundweiss/integrationplugindrexelundweiss.cpp]: Indentation corrected

**Review by mzanetti — APPROVED** [review]
Thank you!

**Comment by mzanetti** [comment]
merged

**Pull request closed without merging**